### PR TITLE
refactor(FOROME-723): modals custom inheritance mode

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-721

PR into refactor/FOROME-722

1. Split select and edit modals into one
2. Added special store for custom inh mode modal

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    switch (type) {
+      case 'first':
+        this.selectValues.first = value
+        break
+      case 'second':
+        this.selectValues.second = value
+        break
+      case 'third':
+        this.selectValues.third = value
+        break
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts]: this.selectValues[type] = value

File: src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+    third: '--',
+  }
+
+  resetValue = ''
+
+  constructor() {
+    makeAutoObservable(this)
+  }
+
+  public setSingleSelectValue(type: string, value: string): void {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts]: type: keyof ISelectValues

File: src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+    return this.selectValues.first
+  }
+
+  get secondSelectValue(): string {
+    return this.selectValues.second
+  }
+
+  get thirdSelectValue(): string {
+    return this.selectValues.third
+  }
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts]: I don't think those getters are necessary

File: src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+
+  public setResetValue(resetValue: string) {
+    this.resetValue = resetValue
+  }
+
+  get reset(): string {
+    return this.resetValue
+  }
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts]: this one too

File: src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+
+  public sendRequest(
+    type: string,
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts]: type: keyof ISelectValues

File: src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+      if (newScenario[index + 1]) scenarioString += ','
+    })
+
+    const params = `{"scenario":{${scenarioString}}}`
+
+    dtreeStore.fetchStatFuncAsync(modalEditStore.groupName, params)
+  }
+
+  public setSingleScenario = (group: string, value: string): void => {
+    const problemGroups: string[] = modalEditStore.problemGroups
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts]: const {problemGroups} = modalEditStore

File: src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+      this.sendRequest('third', value)
+    }
+
+    this.setResetValue('')
+  }
+
+  public setComplexScenario(name: string): void {
+    const problemGroups: string[] = modalEditStore.problemGroups
+
+    if (name === InheritanceModeEnum.HomozygousRecessive_XLinked) {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.store.ts:109]: too many repeats below. Try to move some logic to a method

File: src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.tsx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+import { getResetType } from '@utils/getResetType'
+import { CustomInheritanceModeContent } from '../../../query-builder/ui/custom-inheritance-mode-content'
+import { HeaderModal } from '../../../query-builder/ui/header-modal'
+import { ModalBase } from '../../../query-builder/ui/modal-base'
+import modalEditStore from '../../modal-edit.store'
+import { EditModalButtons } from '../edit-modal-buttons'
+import modalCustomInheritanceModeStore from './modal-custom-inheritance-mode.store'
+
+export const ModalCustomInheritanceMode = observer((): ReactElement => {
+  const ref = useRef(null)
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.tsx]: Use more descriptive variables. Ref what?

File: src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.tsx [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+          problemGroups[1],
+        ),
+        third: modalCustomInheritanceModeStore.getSelectedValue(
+          problemGroups[2],
+        ),
+      })
+
+      const params = `{"scenario":${scenarioString}}`
+
+      dtreeStore.fetchStatFuncAsync(groupName, params)
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-custom-inheritance-mode/modal-custom-inheritance-mode.tsx]: Isn't this logic more suitable for store?

**Review by AndreyUstyumenko — APPROVED** [review]

**Review by AlMaQntr — APPROVED** [review]

**Pull request closed without merging**